app.rb

```diff
@@ -0,0 +1,152 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'books'
+require_relative 'rentals'
+
+class App
+  attr_accessor :list_books, :list_persons
+
+  def initialize
+    @list_books = []
+    @list_persons = []
+  end
+
+  def books
+    if @list_books.empty?
+      puts "\nthere's no books avalaible"
+    else
+      @list_books.each_with_index do |book, index|
+        print "\n#{index + 1}) - "
+        print "author: #{book.author} ~ "
+        print "title: #{book.title}"
+      end
+    end
+  end
+
+  def persons
+    if @list_persons.empty?
+      puts ''
+      puts "there's no person avalaible"
+    else
+      @list_persons.each_with_index do |per, index|
+        print "\n#{index + 1}) - "
+        print "[#{per.class.name}] id: #{per.id} ~ name: #{per.name} ~ age: #{per.age} ~ "
+        print "specialization: #{per.specialization}" if per.methods.include? :specialization
+      end
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a Student(1) or a Teacher(2)?'
+    person_type = check_emptyness(gets.chomp)
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'this value is not correctly'
+    end
+  end
```

Comment:
[OPTIONAL]

I think it will be better if you print a statement that shows that a person has been successfully created so that the user can ascertain that he /she has performed a successful operation 😄,
e.g. puts "Person created successfully"

Reply:
this is added on each of the methods called.
hence when a student or a teacher is created it will be shown as the following
'Created new (teacher or student) with id: }'